numpy/_core/src/multiarray/item_selection.c

```diff
@@ -1064,10 +1064,25 @@ PyArray_Choose(PyArrayObject *ip, PyObject *op, PyArrayObject *out,
              * so the input array is not changed
              * before the error is called
              */
-            flags |= NPY_ARRAY_ENSURECOPY;
+            copy_out = 1;
         }
+
+        if (!PyArray_EquivTypes(dtype, PyArray_DESCR(out))) {
+            copy_out = 1;
+        }
+
         Py_INCREF(dtype);
-        obj = (PyArrayObject *)PyArray_FromArray(out, dtype, flags);
+        if (copy_out) {
+            obj = (PyArrayObject *)PyArray_NewFromDescr(&PyArray_Type,
+                                                        dtype,
+                                                        multi->nd,
+                                                        multi->dimensions,
+                                                        NULL, NULL, 0,
+                                                        (PyObject *)out);
+        }
+        else {
+            obj = (PyArrayObject *)PyArray_FromArray(out, dtype, 0);
```

Comment:
Please _only_ incref here and assign to `obj`.  We rely on that being true now.

Reply:
Sorry, just to be clear, this would be the `dtype`? Made that change.